main.go

```diff
@@ -47,56 +47,78 @@ func main() {
 		os.Exit(1)
 	}
 	cliFlags := parseFlags()
-	doSelectedOperation(cliFlags)
+	exitCode := doSelectedOperation(cliFlags)
+	os.Exit(exitCode)
 }
 
 func parseFlags() (cliFlags cli.Flags) {
 	flag.BoolVar(&cliFlags.ShowConfiguration, "show-configuration", false, "show configuration")
 	flag.BoolVar(&cliFlags.ShowAuthors, "show-authors", false, "show authors")
 	flag.BoolVar(&cliFlags.ShowVersion, "show-version", false, "show version")
+	flag.BoolVar(&cliFlags.InitService, "init-service", false, "initialize the service. Useful for checking all the configuration is right")
```

Comment:
I'm struggling with calling the switch `init-service`. The name makes sense in the context of the code - we separate init tasks from starting a server. I don't think I would understand what it does without looking at the code. 

Instead of changing the names, a more informative description might be enough:

```suggestion
	flag.BoolVar(&cliFlags.InitService, "init-service", false, "perform initialization tasks and stop. Useful for validating the configuration")
```

Similarly in the `Makefile`.

Reply:
solved in https://github.com/RedHatInsights/insights-operator-gathering-conditions-service/pull/266/commits/becdcbac23078e97ee06309199551a8808bd3baf

Reply:
please mark the thread as resolved once checked :)

Reply:
Thanks for renaming the option.

Since the option is now call `check-config`, what information does a user get from the "Initialize the service" part of the help strings? Why don't we say just "Check all the configuration files are parsable" in the help/description strings?

Reply:
solved in https://github.com/RedHatInsights/insights-operator-gathering-conditions-service/pull/266/commits/e739aaaeaf674c12ac59525df935970485fb9e1f. Closing this thread